src/ansys/aedt/core/modeler/cad/primitives.py

```diff
@@ -6280,6 +6281,82 @@ def update_object(self, assignment):
         o = self._create_object(name)
         return o
 
+    @pyaedt_function_handler()
+    def update_geometry_property(self, assignment, name=None, value=None):
+        """Update property of assigned geometry objects.
+
+        Parameters
+        ----------
+        assignment : str, or list
+            Object name or list of object names to be updated.
+        name : str, optional
+            Property name to change. The default is ``None``, in which case no property is updated.
+            Available options are: ``"display_wireframe"``, `"material"``, and `"solve_inside"``.
+        value : bool or str, optional
+            Property value. The default is ``None`` in which case
+            no value is assigned.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        """
+        assignment = self.convert_to_selections(assignment, True)
+
+        # Define property mapping
+        property_mapping = {
+            "display_wireframe": {"property_name": "Display Wireframe", "reset_attr": ["_wireframe"]},
+            "material_name": {"property_name": "Material", "reset_attr": ["_material_name", "_model", "_solve_inside"]},
+            "solve_inside": {"property_name": "Solve Inside", "reset_attr": ["_solve_inside"]},
+            "color": {"property_name": "Color", "reset_attr": ["_color"]},
+            "transparency": {"property_name": "Transparent", "reset_attr": ["_transparency"]},
+            "part_coordinate_system": {"property_name": "Orientation", "reset_attr": ["_part_coordinate_system"]},
+        }
+
+        # Check if property name is valid
+        property_key = name.lower()
+        if property_key not in property_mapping:
+            self.logger.error("Invalid property name.")
+            return False
+
+        # Retrieve property settings
+        property_name = property_mapping[property_key]["property_name"]
+        reset_attr = property_mapping[property_key]["reset_attr"]
+
+        # Handle special cases for material
+        if property_key == "material_name" and isinstance(value, str):
+            matobj = self._materials.checkifmaterialexists(value)
+            if matobj:
+                value = f'"{matobj.name}"'
+            elif "[" in value or "(" in value:  # pragma: no cover
+                value = value
+            else:
+                self.logger.error("Invalid material value.")
+                return False
+
+        value_command = ["Value:=", value]
+        if property_key == "color":
+            if isinstance(value, tuple) or isinstance(value, list):
+                R = clamp(value[0], 0, 255)
+                G = clamp(value[1], 0, 255)
+                B = clamp(value[2], 0, 255)
+                value_command = ["R:=", str(R), "G:=", str(G), "B:=", str(B)]
+            else:
+                self.logger.error("Invalid color.")
+                return False
+
+        # Reset property values
```

Comment:
Is there a reason for resetting the property values before changing them with `_change_geometry_property` ?